python/ccf/receipt.py

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+from hashlib import sha256
+from typing import List
+
+
+def root(leaf: str, proof: List[dict]):
+    current = bytes.fromhex(leaf)
+    for n in proof:
+        if "left" in n:
+            current = sha256(bytes.fromhex(n["left"]) + current).digest()
+        else:
+            current = sha256(current + bytes.fromhex(n["right"])).digest()
+    return current.hex().upper()
```

Comment:
Why `.upper()`? This seems inconsistent with the way we already report hex-encoded strings in CCF.

Reply:
Because the merklecpp built-in does upper, but I'll make that configurable to match the rest of CCF, or even switch the default there.
@wintersteiger do you feel strongly about hex string casing? Do you mind if we change the merklecpp default to match ccf?

Reply:
No, I don't care about the casing at all! Happy to change it to lower-case if you prefer that. The Hash constructor from hex string should accept both as it is. I think I'll add a parameter with default lower-case; might be helpful for other applications.